C/services/notification-plugin-interfaces/python/python_plugin_interface.cpp

```diff
@@ -0,0 +1,657 @@
+/*
+ * FogLAMP south plugin interface related
+ *
+ * Copyright (c) 2019 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Massimiliano Pinto
+ */
+
+#include <logger.h>
+#include <config_category.h>
+#include <reading.h>
+#include <mutex>
+#include <plugin_handle.h>
+#include <Python.h>
+
+#include <python_plugin_common_interface.h>
+
+#define SHIM_SCRIPT_REL_PATH  "/python/foglamp/plugins/common/shim/"
+#define SHIM_SCRIPT_NAME "notification_shim"
+
+using namespace std;
+
+extern "C" {
+
+extern PLUGIN_INFORMATION *Py2C_PluginInfo(PyObject *);
+extern void logErrorMessage();
+extern PLUGIN_INFORMATION *plugin_info_fn();
+extern PLUGIN_HANDLE plugin_init_fn(ConfigCategory *);
+extern void plugin_shutdown_fn(PLUGIN_HANDLE);
+
+// Reconfigure entry point for rule and delivery plugings
+void notification_plugin_reconfigure_fn(PLUGIN_HANDLE,
+					const std::string&);
+// Notification rule plugin entry points
+std::string plugin_triggers_fn(PLUGIN_HANDLE handle);
+bool plugin_eval_fn(PLUGIN_HANDLE handle,
+		    const std::string& assetValues);
+std::string plugin_reason_fn(PLUGIN_HANDLE handle);
+//Notificztion deelivery plugin entry point
+bool plugin_deliver_fn(PLUGIN_HANDLE handle,
+			const std::string& deliveryName,
+			const std::string& notificationName,
+			const std::string& triggerReason,
+			const std::string& customMessage);
+
+// Python object to instantiate
+PyObject* pModule = NULL;
+
+// PluginName
+string sPluginName;
+
+/**
+ * Constructor for PythonPluginHandle
+ *    - Load python 3.5 interpreter
+ *    - Set sys.path and sys.argv
+ *    - Import shim layer script and pass plugin name in argv[1]
+ *    - Set plygin_type (notificationRule or notificationDelivery in rgv[2]
+ *
+ * @param    pluginName		The plugin name to load
+ * @param    pluginPathName	The plugin pathname
+ */
+void *PluginInterfaceInit(const char *pluginName, const char * pluginPathName)
+{
+	// Set plugin name
+	sPluginName = pluginName;
+	// Get FOGLAMP_ROOT dir
+	string foglampRootDir(getenv("FOGLAMP_ROOT"));
```

Comment:
It's unlikely I guess, but if the environment variable is missing you will have an issue here.

Reply:
Yes, next commits